test/Pool.test.js

```diff
@@ -1315,6 +1315,22 @@ describe('Pool', function () {
     });
   });
 
+  it('should not terminate worker if abort listener is defined', function () {
+    var pool = createPool(__dirname + '/workers/cleanup-abort.js');
+    
+    pool.exec('asyncTimeout', [])
+    .timeout(200)
+    .catch((err) => {
+      assert(err instanceof Promise.TimeoutError);
+      // We should still have a busy worker since the worker should still be
+      // active and the promise chain for the timeout has yet to resolve
+      // which keeps the task active.
+      assert(pool.stats().busyWorkers === 1);
```

Comment:
Emitting an event through `worker.emit` is not possible from the `onAbort` context since the task will be cleaned up if `timeout` or `cancel` is invoked by the time the `onAbort` handler is called.

Reply:
@josdejong Do we wish to allow for `emit` to extend to tasks which are being `tracked` as this PR now introduces it as a concept internally for cleanup support. If we do allow lookup in this map based on the `id` in the payload then we could support event emitting during task cleanup.

Reply:
ah, that makes sense. Would it add much complexity to support `emit` from within the abort handler? If so I think it's not worth it.

Reply:
o wait I see I'm replying on an outdated comment, please ignore...